// docusaurus2-dotenv/dist/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var dotenv_webpack_1 = __importDefault(require("dotenv-webpack"));
var DEFAULT_OPTIONS = {
    path: "./.env",
    safe: false,
    systemvars: false,
    silent: false,
    expand: false,
    defaults: false,
};
module.exports = function (context, opts) {
    var options = __assign(__assign({}, DEFAULT_OPTIONS), opts);
    return {
        name: "docusaurus2-dotenv",
        configureWebpack: function (config, isServer) {
            return {
                plugins: [new dotenv_webpack_1.default(options)],
                module: {
                    rules: [],
                },
            };
        },
    };
};
